### PR TITLE
[IMP/Fix]Improved group by functionality.

### PR DESCRIPTION
**Pull request by MeetKD**
1. Fixed the issue of expand shrink buttons showing on discuss menu.
2. Added debounce event when buttons are clicked.
3. Fixed the issue of buttons are added in FavoriteMenu.
4. Restructured the code to fix other various issue like only display group by expand buttons if group by button is visible. 
(NOTE: Earlier code was not honoring **disable_groupby** by flag in searchview)

CCing @pedrobaeza @aldo774 @tarteo

### CODE REVIEW

File: web_group_expand/static/src/js/web_group_expand.js [hunk 1]
```diff
@@ -1,75 +1,120 @@
-odoo.define('web_groupby_expand.web_groupby_expand', function (require) {
-"use strict";
+odoo.define('web.GroupByExpandMenu', function (require) {
```
Comment (simahawk) [inline web_group_expand/static/src/js/web_group_expand.js]: module name should stay as namespace and I'd slit them into 2 files

**Comment by MeetKD** [comment]
@simahawk can you please review this?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: web_group_expand/static/src/js/web_group_expand.js (hunk 1 above)
Comment (MeetKD) [reply web_group_expand/static/src/js/web_group_expand.js]: @simahawk Can you please re-review ?? :smiley:

**Review by aldo774 — APPROVED** [review]
You did exactly what I wanted to do when I migrated this module, I couldn't for lack of experience, I learned by seeing your code, thanks.

**Comment by MeetKD** [comment]
@aldo774 i am glad  and thank you for your kind words. :green_heart:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**